### ISSUE TITLE
[BUG] Authentication failed against database server at `postgres`, the provided database credentials for `ghostfolio` are not valid

### ISSUE THREAD

**Issue by rubengr** [ghostfolio/ghostfolio#3889]
Bug Description

Try to run docker image, get the error **Error: P1000: Authentication failed against database server at `postgres`, the provided database credentials for `ghostfolio` are not valid.**

I just changed the .env file according readme file.

Screenshots

![image](https://github.com/user-attachments/assets/523e6327-7566-4c07-a0b2-d347f3026fee)

![image](https://github.com/user-attachments/assets/0cca3174-b2c9-47cc-9ec1-e75229a17dce)

Environment

Ghostfolio Version latest
Self-hosted
Experimental Features enabled
Browser Chrome
OS Windows 11 Home

**Comment by dtslvr** [comment]
Hi @rubengr

Have you started the database container before changing the `.env` file?

**Comment by rubengr** [comment]
Hi,
No, I already had a dev container but it was stopped. Now I ran compose from dev example and it worked.

Is normal this error?

![image](https://github.com/user-attachments/assets/41aaf419-6e06-466f-a6c4-f5cb5b2d5ffb)

**Comment by dtslvr** [comment]
> No, I already had a dev container but it was stopped.

It's likely related to the volume still holding the old credentials. Try stopping the containers, removing the volume, and then starting them again after updating the `.env` file. I'm closing this issue as it does not seem to be a bug.

> Is normal this error?

Yes, please check the [FAQ](https://ghostfol.io/en/faq/self-hosting) to find out how to resolve it.

**Comment by pippo73** [comment]
Hello @dtslvr 
I'm sorry but I'm having the same problem. 

here are my files:

**docker-compose.yml**
```
services:
  ghostfolio:
    image: docker.io/ghostfolio/ghostfolio:latest
    init: true
    cap_drop:
      - ALL
    security_opt:
      - no-new-privileges:true
    env_file:
      - ./.env
    environment:
      DATABASE_URL: postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@postgres:5432/${POSTGRES_DB}?connect_timeout=300&sslmode=prefer
      REDIS_HOST: redis
      REDIS_PASSWORD: ${REDIS_PASSWORD}
    ports:
      - 3333:3333
    depends_on:
      postgres:
        condition: service_healthy
      redis:
        condition: service_healthy
    healthcheck:
      test: ['CMD-SHELL', 'curl -f http://localhost:3333/api/v1/health']
      interval: 10s
      timeout: 5s
      retries: 5

  postgres:
    image: docker.io/library/postgres:15
    cap_drop:
      - ALL
    cap_add:
      - CHOWN
      - DAC_READ_SEARCH
      - FOWNER
      - SETGID
      - SETUID
    security_opt:
      - no-new-privileges:true
    env_file:
      - ./.env
    healthcheck:
      test: ['CMD-SHELL', 'pg_isready -d $${POSTGRES_DB} -U $${POSTGRES_USER}']
      interval: 10s
      timeout: 5s
      retries: 5
    volumes:
      - postgres:/var/lib/postgresql/data

  redis:
    image: docker.io/library/redis:alpine
    user: '999:1000'
    cap_drop:
      - ALL
    security_opt:
      - no-new-privileges:true
    env_file:
      - ./.env
    command: ['redis-server', '--requirepass', $REDIS_PASSWORD]
    healthcheck:
      test: ['CMD-SHELL', 'redis-cli --pass $REDIS_PASSWORD ping | grep PONG']
      interval: 10s
      timeout: 5s
      retries: 5

volumes:
  postgres:
```

**.env**
```
COMPOSE_PROJECT_NAME=ghostfolio

# CACHE
REDIS_HOST=localhost
REDIS_PORT=6379
REDIS_PASSWORD=RNYznLuA9Zn

# POSTGRES
POSTGRES_DB=ghostfolio-db
POSTGRES_USER=user
POSTGRES_PASSWORD=duA9CShvonDE

# VARIOUS
ACCESS_TOKEN_SALT=b!ZNq4<u4+uP52
DATABASE_URL=postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@localhost:5432/${POSTGRES_DB}?connect_timeout=300&sslmode=prefer
JWT_SECRET_KEY==24iLzXeiCnXQD

```

this are my commands to be sure that I have no traces of previous ghost 

```
$ docker system prune
WARNING! This will remove:
  - all stopped containers
  - all networks not used by at least one container
  - all dangling images
  - unused build cache

Are you sure you want to continue? [y/N] y
Total reclaimed space: 0B


docker ps -a
CONTAINER ID   IMAGE                                           COMMAND                  CREATED             STATUS         PORTS                                       NAMES
a96548a57656   revenz/fenrus                                   "/app/docker-entrypo…"   49 minutes ago      Up 6 minutes   0.0.0.0:3000->3000/tcp, :::3000->3000/tcp   fenrus
8a0949c1a33a   lscr.io/linuxserver/changedetection.io:latest   "/init"                  About an hour ago   Up 6 minutes   0.0.0.0:5000->5000/tcp, :::5000->5000/tcp   changedetection
5a0f47327126   portainer/agent                                 "./agent"                3 hours ago         Up 6 minutes   0.0.0.0:9001->9001/tcp, :::9001->9001/tcp   portainer_agent

$ docker rmi ghostfolio/ghostfolio:latest postgres:15 redis:alpine
Error response from daemon: No such image: ghostfolio/ghostfolio:latest
Error response from daemon: No such image: postgres:15
Error response from daemon: No such image: redis:alpine
```

Now that I'm sure that there are not "problems" I start ghostfolio docker

```
$ docker compose up -d
[...]
[+] Running 4/4
 ✔ Network ghostfolio_default         Created                                                                                                            0.1s
 ✔ Container ghostfolio-redis-1       Healthy                                                                                                           22.6s
 ✔ Container ghostfolio-postgres-1    Healthy                                                                                                           22.6s
 ✔ Container ghostfolio-ghostfolio-1  Started 
```

and here is the problem

```
$ docker ps
CONTAINER ID   IMAGE                                           COMMAND                  CREATED              STATUS                        PORTS                                       NAMES
88910acadd1f   redis:alpine                                    "docker-entrypoint.s…"   About a minute ago   Up About a minute (healthy)   6379/tcp                                    ghostfolio-redis-1
b3c57468d62f   postgres:15                                     "docker-entrypoint.s…"   About a minute ago   Up About a minute (healthy)   5432/tcp                                    ghostfolio-postgres-1
a96548a57656   revenz/fenrus                                   "/app/docker-entrypo…"   56 minutes ago       Up 13 minutes                 0.0.0.0:3000->3000/tcp, :::3000->3000/tcp   fenrus
8a0949c1a33a   lscr.io/linuxserver/changedetection.io:latest   "/init"                  About an hour ago    Up 13 minutes                 0.0.0.0:5000->5000/tcp, :::5000->5000/tcp   changedetection
5a0f47327126   portainer/agent                                 "./agent"                3 hours ago          Up 13 minutes                 0.0.0.0:9001->9001/tcp, :::9001->9001/tcp   portainer_agent

$ docker logs ghostfolio-ghostfolio-1 -f
+ echo Running database migrations
Running database migrations
+ npx prisma migrate deploy
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "ghostfolio-db", schema "public" at "postgres:5432"

Error: P1000: Authentication failed against database server at `postgres`, the provided database credentials for `user` are not valid.

Please make sure to provide valid database credentials for the database server at `postgres`.
npm notice
npm notice New minor version of npm available! 10.8.2 -> 10.9.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v10.9.0
npm notice To update run: npm install -g npm@10.9.0
npm notice

$ docker logs ghostfolio-postgres-1 -f

PostgreSQL Database directory appears to contain a database; Skipping initialization

2024-10-16 09:50:58.739 UTC [1] LOG:  starting PostgreSQL 15.8 (Debian 15.8-1.pgdg120+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
2024-10-16 09:50:58.739 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2024-10-16 09:50:58.739 UTC [1] LOG:  listening on IPv6 address "::", port 5432
2024-10-16 09:50:58.740 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2024-10-16 09:50:58.743 UTC [28] LOG:  database system was shut down at 2024-10-16 09:44:55 UTC
2024-10-16 09:50:58.747 UTC [1] LOG:  database system is ready to accept connections
2024-10-16 09:51:10.200 UTC [40] FATAL:  password authentication failed for user "user"
2024-10-16 09:51:10.200 UTC [40] DETAIL:  Connection matched pg_hba.conf line 100: "host all all all scram-sha-256"

```


Do I miss something? where is the problem?

**Comment by dtslvr** [comment]
@pippo73 I’d suggest to start a new discussion to get more visibility.
